design/specs/foreign-key-constraints.md

```diff
@@ -2,75 +2,209 @@
-For users, this knowledge of "linked tables" is critical for the system's overall usability.
+### Videos
 
-### Primary Key
-Primary keys are unique identifiers used to identify records in a table. In Mathesar, primary keys are set by default to the id column, and their values are generated automatically. 
+----
 
-### Foreign Key
-The foreign key is a relationship between two tables that identifies which column in the other table references the first one.
+- Scenario 1: <https://www.loom.com/share/01fb12b24f0e4270822da212622a8844>
+- Scenario 2: <https://www.loom.com/share/87fce49545c744d1b475693b7d43c2bf>
+- Scenario 3: <https://www.loom.com/share/e43763b172e449fa9fddff25b311123f>
+- Scenario 4: <https://www.loom.com/share/6deed11141f34c4ba495c4adb7117982>
+- Scenario 5: <https://www.loom.com/share/b90ee712055a4a9cb24a499f943c0bbe>
+- Scenario 6: <https://www.loom.com/share/4b617af642244946be965d609b1aed99>
 
-### Lookup Column
-They are used to show information from linked records in another table. The lookup column setting determines which column values are visible when displaying the record.
+----
 
 ## Scenarios
-### A user wants to create a one-to-many table relationship between two existing tables
-A user has a table named 'artist' that contains records for artists in a music album collection. The user wants to link each artist to records in another table named 'track.' If successful, the user should link all tracks associated with an artist to records in the 'artist' table.
 
+### Scenario 1: User Adds a Foreign Key Constraint
 
-### Steps
+A user wants to link records from another table into their current table. Both tables belong to the same schema. For this purpose, foreign key constraints can be applied to change the relationship between one table and another based on the values of columns of the linked tables.
 
-#### 1. The user opens the 'track' table and adds a new column to link records from the 'artist' table
-- From the sidebar navigation list of tables, click on a table named 'track.'
-- Click on 'Add new Column,' and a new column is added with text type by default
-#### The user opens the new column menu and selects the 'Link to Another Table' option
-- Click on the column header to show the menu and select the 'Link to Another Table' option
+### Scenario 1a: Automatically adding foreign key constraints from the 'Link Table' dialog
 
-**Note: This step does not cover the scenario where a user has a unique table, in which case they need to create an additional table to add a relationship.**
+#### Assumptions for 1a
 
-#### 2. The user selects a table from the list of available tables
-- The user selects the table on the 'one side of the 'one-to-many' relationship.
+- The user is not familiar with the concept of foreign keys.
+- The user has created tables within Mathesar, where the ID column is automatically generated and sent as the primary key.
```

Comment:
I don't think this is a good assumption to make. We should handle cases for all tables, even without primary keys.

We can solve this for the first version by not supporting creating links through scenario 1a. Perhaps we could have a little help message that says "Why don't I see all my tables in this list?" and explain that we only show tables with primary keys.

Reply:
I think it would be better if we show all the tables, but disable the table without a primary key with a text hint that would direct the user to create a primary key if they wish to link it to that particular table.

Reply:
I agree with @silentninja. I think it would be best to show the tables and prevent selection. I will add details for this in the spec.